java/code/src/com/redhat/rhn/domain/kickstart/crypto/CryptoKey.java

```diff
@@ -17,18 +17,46 @@
 import com.redhat.rhn.domain.Identifiable;
 import com.redhat.rhn.domain.org.Org;
 
+import org.hibernate.annotations.DiscriminatorFormula;
+
 import java.nio.charset.StandardCharsets;
 
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.SequenceGenerator;
+import javax.persistence.Table;
+
 /**
  * CryptoKey - Class representation of the table rhnCryptoKey.
  */
+@Entity
+@Table(name = "rhnCryptoKey")
+@DiscriminatorFormula("(SELECT type.label FROM rhnCryptoKeyType type WHERE type.id = crypto_key_type_id)")
 public class CryptoKey implements Identifiable {
```

Comment:
I wonder if you need to set `@Inheritance(strategy = InheritanceType.SINGLE_TABLE)` ?
Or is the `@DiscriminatorFormula` enough?